regolith/helpers/l_appointmentshelper.py

```diff
@@ -0,0 +1,140 @@
+    """
+    # btype must be the same as helper target in helper.py
+    btype = HELPER_TARGET
+    needed_dbs = [f'{TARGET_COLL}']
+
+    def construct_global_ctx(self):
+        """Constructs the global context"""
+        super().construct_global_ctx()
+        gtx = self.gtx
+        rc = self.rc
+        if "groups" in self.needed_dbs:
+            rc.pi_id = get_pi_id(rc)
+        rc.coll = f"{TARGET_COLL}"
+        try:
+            if not rc.database:
+                rc.database = rc.databases[0]["name"]
+        except:
+            pass
+        colls = [
+            sorted(
+                all_docs_from_collection(rc.client, collname), key=_id_key
+            )
+            for collname in self.needed_dbs
+        ]
+        for db, coll in zip(self.needed_dbs, colls):
+            gtx[db] = coll
+        gtx["all_docs_from_collection"] = all_docs_from_collection
+        gtx["float"] = float
+        gtx["str"] = str
+        gtx["zip"] = zip
+
+    def sout(self):
+        rc = self.rc
+        # This if statement should be in all listers. Make sure to change self.gtx to get the database the lister needs
+        if rc.filter:
+            results = search_collection(self.gtx["projecta"], rc.filter, rc.keys)
+            print(results, end="")
+            return
+        all_milestones = []
+        if not rc.stati:
+            rc.stati = ['started']
+        for projectum in self.gtx["projecta"]:
+            if rc.lead and projectum.get('lead') != rc.lead:
+                continue
+            projectum["deliverable"].update({"name": "deliverable",
+                                             "objective": "deliver"})
+            projectum["kickoff"].update({"type": "meeting"})
+            gather_miles = [projectum["kickoff"], projectum["deliverable"]]
+            gather_miles.extend(projectum["milestones"])
+            for ms in gather_miles:
+                if projectum["status"] in rc.stati or \
+                        'all' in rc.stati:
+                    if ms.get('status') not in \
+                            ["finished", "cancelled"]:
+                        due_date = get_due_date(ms)
+                        ms.update({
+                            'lead': projectum.get('lead'),
+                            'group_members': projectum.get('group_members'),
+                            'collaborators': projectum.get('collaborators'),
+                            'id': projectum.get('_id'),
+                            'due_date': due_date,
+                            'log_url': projectum.get('log_url'),
+                            'pi': projectum.get('pi_id')
+                        })
+                        all_milestones.append(ms)
+        all_milestones.sort(key=lambda x: x['due_date'], reverse=True)
+        for ms in all_milestones:
+            if rc.verbose:
+                print(
+                    f"{ms.get('due_date')}: lead: {ms.get('lead')}, {ms.get('id')}, status: {ms.get('status')}")

```

Comment:
this doesn't seem to match the test.  I think the first thing should be ``ms.get('id')`` or sthg.